### ISSUE TITLE
"Link" Examples/Proofs to Theorems

### ISSUE THREAD

**Issue by ParadaCarleton** [Marmare314/lemmify#11]
The `use-proof-numbering` function works, but I think a better approach might be to pass a link/reference (as in `@theorem`) from a proof or example to its "parent" theorem. This approach seems cleaner, and also works better in situations where there are multiple proofs or examples corresponding to a single theorem.

**Comment by Marmare314** [comment]
I agree and already started working on this in the `link-to` branch some time ago. But since the current implementation stores values in parameters of figures this will require a bit of a refactoring. The only way I could think of to do this would be to store the parameters in the numbering function instead, which is kind of hacky, but should work. Unfortunately I haven't had time to finish this yet.

**Comment by Marmare314** [comment]
Just merged the link-to parameter into main #17. But because I have not actually tested the use you described of linking to theorems using labels, I am going to leave this issue open for now.

**Comment by ParadaCarleton** [comment]
Thank you! 

> Just merged the link-to parameter into main #17. But because I have not actually tested the use you described of linking to theorems using labels, I am going to leave this issue open for now.

Did you happen to tag a new release? I can't get 0.2.0 in the Typst web app.

**Comment by Marmare314** [comment]
It's not released yet, I need to improve the documentation a bit more and I wanted to get it working with showybox first. Should not be too long though.
